test/specs/rules/disallow-quoted-keys-in-objects.js

```diff
@@ -67,4 +67,12 @@ describe('rules/disallow-quoted-keys-in-objects', function() {
 
         assert(checker.checkString('var x = { "null": 1, undefined: "foo" }').isEmpty());
     });
+
+    it('should report non-reserved quoted keys when "allButReserved" mode is used. #1669', function() {
```

Comment:
We have some tests for this option now, i think it would be logical to move it into its own `describe`  and move
```js
checker = new Checker();
checker.registerDefaultRules();
checker.configure({ disallowQuotedKeysInObjects: 'allButReserved' });
```
bit there too